earthmover/yaml_parser.py

```diff
@@ -30,6 +30,31 @@ def update(self, _dict: 'YamlMapping'):
                 self[key] = val
         return self
 
+    # enables setting values inside the YamlMapping by path, e.g. `configs.set_path("config.tmp_dir", "/mypath")`
+    def set_path(self, path, value):
+        path_pieces = path.split(".")
+        if len(path_pieces)==1:
+            self[path] = self.autocast(value)
+        else:
+            self[path_pieces[0]].set_path(".".join(path_pieces[1:]), value)
+    
+    @staticmethod
+    def autocast(value):
+        if value.lower() in ['true', 'yes', 'on']:
+            return True
+        elif value.lower() in ['false', 'no', 'off']:
+            return False
+        elif '.' in value:
+            try:
+                return float(value)
+            except ValueError:
+                return value
+        else:
+            try:
+                return int(value)
+            except ValueError:
+                return value
```

Comment:
Values sent in via a CLI override are all strings; this helper function casts them to bool, int, float (as appropriate, like the YAML parser does) to enable overriding all "simple" data types.

Reply:
We should include 't'/'f' and 'y'/'n' in the boolean lists to be comprehensive.

Reply:
Good suggestion! I've added those. Marking this resolved.